pkg/providers/imagefamily/ubuntu_2004.go

```diff
@@ -0,0 +1,131 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package imagefamily
+
+import (
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1beta1"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily/bootstrap"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily/customscriptsbootstrap"
+	types "github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily/types"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/launchtemplate/parameters"
+
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	"sigs.k8s.io/karpenter/pkg/scheduling"
+)
+
+const (
+	Ubuntu2004Gen2FIPSImageDefinition = "2004gen2fipscontainerd"
+	Ubuntu2004Gen1FIPSImageDefinition = "2004fipscontainerd"
+)
+
+type Ubuntu2004 struct {
+	Options *parameters.StaticParameters
+}
+
+func (u Ubuntu2004) Name() string {
+	return v1beta1.UbuntuImageFamily
+}
+
+func (u Ubuntu2004) DefaultImages() []types.DefaultImageOutput {
+	// Ubuntu2004 doesn't have default node images (only FIPS)
+	return []types.DefaultImageOutput{}
+}
+
+// FIPS images aren't supported in public galleries, only shared image galleries
+func (u Ubuntu2004) FIPSImages() []types.DefaultImageOutput {
```

Comment:
When you merge with main and the `DefaultImages` has a `sig` parameter, if you take the "do it all with one method" approach you can even have defense-in-depth here where you say:
```
if !fips {
   // return nothing
}

if !useSig {
    // return nothing
}

// return the fips images
```